.github/workflows/kb_sdk_test.yaml

```diff
@@ -0,0 +1,68 @@
+name: KBase SDK Tests
+
+on:
+  push:
+    branches:
+    - master
+    - main
+  pull_request:
+    branches:
+    - master
+    - main
+    - develop
+
+jobs:
+
+  sdk_tests:
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Check out GitHub repo
+      if: "!contains(github.event.head_commit.message, 'skip ci')"
+      uses: actions/checkout@v2
+
+    - name: Check out Actions CI files
+      if: "!contains(github.event.head_commit.message, 'skip ci')"
+      uses: actions/checkout@v2
+      with:
+        repository: 'kbaseapps/kb_sdk_actions'
+        path: 'kb_sdk_actions'
+
+
+    - name: Set up test environment
+      if: "!contains(github.event.head_commit.message, 'skip ci')"
+      shell: bash
+      env:
+        KBASE_TEST_TOKEN: ${{ secrets.KBASE_CI_TOKEN }}
+      run: |
+        # Verify kb_sdk_actions clone worked
+        test -f "$HOME/kb_sdk_actions/bin/kb-sdk" && echo "CI files cloned"
+        # Pull kb-sdk & create startup script
+        docker pull kbase/kb-sdk
+       
+        sh $GITHUB_WORKSPACE/kb_sdk_actions/bin/make_testdir && echo "Created test_local"
+        test -f "test_local/test.cfg" && echo "Confirmed config exists"
+
+    - name: Configure authentication
+      if: "!contains(github.event.head_commit.message, 'skip ci')"
+      shell: bash
+      env:
+        KBASE_TEST_TOKEN: ${{ secrets.KBASE_CI_TOKEN }}
+        KBASE_TEST_TOKEN2: ${{ secrets.KBASE_CI_TOKEN2 }}
+        KBASE_ENDPOINT_ESCAPED: ${{ 'https:\/\/ci.kbase.us\/services' }}
+        KBASE_AUTH_SERVICE_URL_ESCAPED: ${{ 'https:\/\/ci.kbase.us\/services\/auth\/api\/legacy\/KBase\/Sessions\/Login' }}
+      run: |
+        # Add token to config
+        sed -ie "s/^test_token=.*$/&$KBASE_TEST_TOKEN/g" ./test_local/test.cfg
+        echo test_user2_token=$KBASE_TEST_TOKEN2 >> ./test_local/test.cfg
+        sed -ie "s/^kbase_endpoint=.*$/kbase_endpoint=$KBASE_ENDPOINT_ESCAPED/g" ./test_local/test.cfg
+        sed -ie "s/^auth_service_url=.*$/auth_service_url=$KBASE_AUTH_SERVICE_URL_ESCAPED/g" ./test_local/test.cfg
+
+    - name: Run tests
+      if: "!contains(github.event.head_commit.message, 'skip ci')"
+      shell: bash
+      env:
+        CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
```

Comment:
Does codecov require a token now? In travis it didn't

Reply:
It seems like codecov isn't working though, I don't see a coverage report in the PR

Reply:
These lines come from Json's [template](https://github.com/kbaseapps/kb_sdk_actions/blob/master/action.yaml), I am not very sure about the token thing.
The codecov is at the bottom of run tests of sdk_tests: https://github.com/kbaseapps/HTMLFileSetServ/runs/2124974739?check_suite_focus=true#step:6:4086

Reply:
Yeah, it looks like coverage isn't set up: https://github.com/kbaseapps/HTMLFileSetServ/runs/2123859288#step:6:4113

I would just comment out the codecov line and add a TODO saying it needs to be added at some point to avoid confusion.

Reply:
ok!